From 1c5b7848e1a85e99afb5b9ecfe56dc518e2d1735 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Mon, 2 Jun 2014 09:52:03 +0200
Subject: [PATCH] dracut: precreate initqueue hook dirs

modules might want to use the initqueue-timeout to install hooks
---
 dracut                   | 3 ++-
 modules.d/99base/install | 5 -----
 2 files changed, 2 insertions(+), 6 deletions(-)

diff --git a/dracut b/dracut
index 316a6a8..cb79dbf 100755
--- a/dracut
+++ b/dracut
@@ -225,7 +225,8 @@ if [[ -f "$outfile" ]] &&  ! [[ -w "$outfile" ]]; then
     exit 1
 fi
 
-hookdirs="cmdline pre-udev pre-trigger netroot pre-mount pre-pivot mount emergency"
+hookdirs="cmdline pre-udev pre-trigger netroot pre-mount pre-pivot mount emergency initqueue initqueue-finished initqueue-settled initqueue-timeout
+"
 
 [[ -n "$TMPDIR" ]] && ! [[ -w "$TMPDIR" ]] && unset TMPDIR
 readonly initdir=$(mktemp -d -t initramfs.XXXXXX)
diff --git a/modules.d/99base/install b/modules.d/99base/install
index 51fa131..4a4531a 100755
--- a/modules.d/99base/install
+++ b/modules.d/99base/install
@@ -16,11 +16,6 @@ fi
 inst "$moddir/init" "/init"
 inst "$moddir/initqueue" "/sbin/initqueue"
 inst "$moddir/loginit" "/sbin/loginit"
-mkdir -p ${initdir}/initqueue
-mkdir -p ${initdir}/initqueue-finished
-mkdir -p ${initdir}/initqueue-settled
-mkdir -p ${initdir}/initqueue-timeout
-mkdir -p ${initdir}/tmp
 # Bail out if switch_root does not exist
 if which switch_root >/dev/null 2>&1; then
     dracut_install switch_root 
